subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/DisambiguateArtifactTransformIntegrationTest.groovy

```diff
@@ -397,4 +398,73 @@ task resolveTestClasses {
         and:
         output.count("Transforming") == 0
     }
+
+    @Issue("gradle/gradle#8363")
+    def "attribute compatible transform but insufficient will not be selected"() {
+        def m1 = mavenRepo.module("test", "test", "1.3").publish()
+        m1.artifactFile.text = "1234"
+
+        given:
+        buildFile << """
+def artifactType = Attribute.of('artifactType', String)
+def minified = Attribute.of('minified', Boolean)
+
+apply plugin: 'java'
+
+allprojects {
+    repositories {
+        maven { url "${mavenRepo.uri}" }
+    }
+}
+
+dependencies {
+    implementation 'test:test:1.3'
+    
+    artifactTypes.getByName("jar") {
+        attributes.attribute(minified, false)
+    }
+    
+    registerTransform {
+        from.attribute(artifactType, 'jar')
+        to.attribute(artifactType, 'size')
+        artifactTransform(FileSizer)
+    }
+    
+    registerTransform {
+        from.attribute(minified, false)
+        to.attribute(minified, true)
+        artifactTransform(FileSizer)
```

Comment:
❌ Can we add a different transform type for minification? It may have same side effects to use the same type here.